src/sentry/api/serializers/models/group.py

```diff
@@ -173,46 +172,267 @@ def __init__(
+            "isPublic": share_id is not None,
+            "platform": obj.platform,
+            "project": {
+                "id": str(obj.project.id),
+                "name": obj.project.name,
+                "slug": obj.project.slug,
+                "platform": obj.project.platform,
+            },
+            "type": obj.get_event_type(),
+            "metadata": obj.get_event_metadata(),
+            "numComments": obj.num_comments,
+            "assignedTo": serialize(attrs["assigned_to"], user, ActorSerializer()),
+            "isBookmarked": attrs["is_bookmarked"],
+            "isSubscribed": is_subscribed,
+            "subscriptionDetails": subscription_details,
+            "hasSeen": attrs["has_seen"],
+            "annotations": attrs["annotations"],
+        }
+
+        # This attribute is currently feature gated
+        if "is_unhandled" in attrs:
+            group_dict["isUnhandled"] = attrs["is_unhandled"]
+        if "times_seen" in attrs:
+            group_dict.update(self._convert_seen_stats(attrs))
+
+        # XXX(gilbert): remove this after performance issues internal testing
+        from sentry import features
+
+        if features.has("projects:performance-issue-details-backend", obj.project):
+            group_dict["issue_category"] = obj.get_issue_category()
+            group_dict["issue_type"] = obj.type
+
+        return group_dict
+
+    def _get_seen_stats(self, item_list, user) -> Mapping[Any, SeenStats]:
         """
         Returns a dictionary keyed by item that includes:
             - times_seen
             - first_seen
             - last_seen
             - user_count
         """
-        raise NotImplementedError
+        # partition the item_list by type
+        error_issues = [group for group in item_list if "error" == group.get_issue_category()]
+        perf_issues = [group for group in item_list if "performance" == group.get_issue_category()]
 
-    @staticmethod
-    def _get_start_from_seen_stats(seen_stats):
-        # Try to figure out what is a reasonable time frame to look into stats,
-        # based on a given "seen stats".  We try to pick a day prior to the earliest last seen,
-        # but it has to be at least 14 days, and not more than 90 days ago.
-        # Fallback to the 30 days ago if we are not able to calculate the value.
-        last_seen = None
-        if seen_stats:
-            for item in seen_stats.values():
-                if last_seen is None or (item["last_seen"] and last_seen > item["last_seen"]):
-                    last_seen = item["last_seen"]
+        # bulk query for the seen_stats by type
+        error_stats = self._seen_stats_error(error_issues, user) or {}
+        perf_stats = (self._seen_stats_performance(perf_issues, user) if perf_issues else {}) or {}
+        agg_stats = {**error_stats, **perf_stats}
+        # combine results back
+        return {group: agg_stats.get(group, {}) for group in item_list}
```

Comment:
This seems to be the only performance-issue specific thing we need to update, besides the `issue_type` and `issue_category`